doc/sag/sag-HLD.md

```diff
@@ -0,0 +1,185 @@
+Anycast gateway feature for EVPN/VXLAN is a default gateway addressing mechanism that enables the use of the same gateway IP address across all the leaf switches that are part of a VXLAN network.  
+This ensures that every leaf switch can function as the default gateway for the workloads directly connected to it.  
+The feature failitates flexible workload placement, host mobility and optimal traffic forwarding across the VXLAN fabric.  
+
+This feature should be used together with VxLAN EVPN.  
+It should always to configure VxLAN to make SAG work.  
+SAG announces its IP and MAC only towards the host facing Ethernet ports but not towards the fabric (Overlay).  
+As a result, the IP/MAC for the SAG is only seen and reachable from the host facing side and never from the fabric (Overlay) facing side.
+
+The following is the requirements:  
+* Support one global virtual MAC address on SAG interface
+* Support multiple IPv4/IPv6 address on SAG interface
+* Support global IPv4/IPv6 enable knob for SAG address
+
+# Modules Design
+## Config DB
+### SAG Global Table
+```
+SAG_GLOBAL|IP 
+  "gwmac": {{mac_address}} 
+  "IPv4": {{enable}}
+  "IPv6": {{enable}}
+```
+### SAG Interface Table
+```
+SAG|{{vlan_intf_name}}|IPv4
+  "gwip@": {{ipv4_list}}
+SAG|{{vlan_intf_name}}|IPv6
+  "gwip@": {{ipv6_ilst}}
+```
+
+### Config DB Schemas
+```
+; Defines schema for SAG configuration attributes
+key                                   = SAG_GLOBAL|IP                 ; SAG global configuration
+; field                               = value
+gwmac                                 = mac_address                   ; global anycast gateway MAC address
+IPv4                                  = enable/disable                ; enable anycast gateway IPv4 address or not
+IPv6                                  = enable/disable                ; enable anycast gateway IPv6 address or not
+	  
+```
+## APP DB
+### Interface Table
+```
+INTF_TABLE:{{intf_name}}
+  "mac_addr": {{mac_address}}
+
+INTF_TABLE:{{intf_name}}:{{ip_prefix}}
+  "scope": {{addr_scope}
+  "family": {{ip_family}}
+```
+### APP DB Schemas
+```
+; Defines schema for SAG configuration attributes
+key                                   = INTF_TABLE:sag_name           ; SAG interface 
+; field                               = value
+mac_addr                              = mac_address                   ; global anycast mac MAC address
+
+key                                   = INTF_TABLE:sag_name:ip_prefix ; SAG IP address
+; field                               = value
+scope                                 = global/local/host             ; IP address scope
+family                                = IPv4/IPv6                     ; IP address family
+```
+
+## Orchestration Agent
+Following orchagents shall be modified. Flow diagrams are captured in a later section. 
+
+![SAG arch](./images/sag-db-arch.png)
+
+### SagMgrD
+SagMgrD creates the macvlan kernel interfaces on top of the vlan interface.  
```

Comment:
Why cannot the SAG IP address be configured on the SVI interface itself? Note that when only SAG IP address is configured by the user, it will have to be configured on the macvlan netdevice (and not on the parent SVI netdevice). Because of this, ARP/ND will be learnt on the macvlan interface. In such a case, neighsyncd will have to listen to updates on both macvlan and SVI. Similarly, EVPN control plane will have to install remote ARP/ND on the macvlan netdevice.

The macvlan suits well for VRRP use-case where there is a real IP (on SVI) and virtual IP (on macvlan). And ARPs are resolved on the SVI interface.

Reply:
Use the SVI interface instead of using macvlan interface, SAG directly change the SVI interface's MAC address.  
Please refer to the updated HLD.